submission_runner.py

```diff
@@ -315,6 +315,7 @@ def train_once(
   metrics = {'eval_results': eval_results, 'global_step': global_step}
   if USE_PYTORCH_DDP:
     # Sync final score (accumulated training time); choose highest, i.e. worst.
+    dist.barrier()
```

Comment:
It surely doesn't hurt, but out of curiosity, did you see any issue without the barrier here?

Reply:
@rakshithvasudev had an issue with it and was solved after adding a barrier here.

Reply:
Is without that the summary was deadlocked :)

Reply:
Ok interesting, thanks for fixing it!